src/components/AddTask/AddTask.tsx

```diff
@@ -1,11 +1,169 @@
+import DICTIONARY from './addTask.dictionary';
+
+const useStyles = makeStyles(theme => ({
+  wrapper: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+  },
+  header: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    margin: theme.spacing(1, 0, 2),
+  },
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+  },
+  inputWrapper: {
+    flex: 1,
+  },
+  addButton: {
+    margin: theme.spacing(5, 0, 3),
+    fontWeight: 'bold',
+  },
+}));
+
+const AddTask: React.FC<AddTaskProps> = ({ history, showSnackbar }) => {
+  const { wrapper, header, addButton, inputWrapper, form } = useStyles();
+  const { newbieId } = useParams<QueryNewbieArgs>();
+  const { register, errors, handleSubmit, reset } = useForm<TaskInput>();
+  const { state } = useLocation();
+
+  const onCompleted = () => {
+    showSnackbar(DICTIONARY.SUCCESS_MESSAGE);
+    reset();
+  };
+
+  const [
+    addBuddyTask,
+    { loading: addBuddyTaskLoading, error: addBuddyTaskError },
+  ] = useMutation<Mutation>(ADD_BUDDY_TASK, { onCompleted });
+
+  const [
+    addNewbieTask,
+    { loading: addNewbieTaskLoading, error: addNewbieTaskError },
+  ] = useMutation<Mutation>(ADD_NEWBIE_TASK, { onCompleted });
+
+  const addTaskHandlers = [addNewbieTask, addBuddyTask];
+  const defaultTabIndex = (state && state.tabIndex) || 0;
+  const loading = addBuddyTaskLoading || addNewbieTaskLoading;
+  const taskListPath = ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', newbieId);
+
+  const onSubmit = (input: TaskInput) => {
+    addTaskHandlers[defaultTabIndex]({
+      variables: {
+        newbieId,
+        input: {
+          ...input,
+          description: xss.sanitize(input.description || ''),
+        },
+      },
+    });
+  };
+
+  const onBackClick = () =>
+    history.push({ pathname: taskListPath, state: { defaultTabIndex } });
+
+  useEffect(() => {
+    if (addBuddyTaskError || addNewbieTaskError) {
+      showSnackbar(DICTIONARY.ERROR_MESSAGE);
+    }
+  }, [addBuddyTaskError, addNewbieTaskError, showSnackbar]);
+
+  const renderAddTask = () => (
+    <Box className={wrapper} data-testid='task-details-page'>
+      <Box className={header}>
+        <Typography component='h2' variant='h2'>
+          {DICTIONARY.ADD_TASK_TITLE}
+        </Typography>
+      </Box>
+      <form
+        className={form}
+        data-testid='form'
+        noValidate
+        onSubmit={handleSubmit(onSubmit)}>
+        <Box className={inputWrapper}>
+          <TextField
+            inputProps={{ 'data-testid': 'title' }}
+            inputRef={register({
+              required: DICTIONARY.TITLE.REQUIRED,
+            })}
+            margin={'normal'}
+            fullWidth
+            label={DICTIONARY.TITLE.LABEL}
+            name='title'
+            autoFocus
+            error={!!errors.title}
+            helperText={errors.title && errors.title.message}
+          />
+          <TextField
```

Comment:
@Tonik1204  please use `helperText` for brief explanation of the field.
`helperText="HTML syntax is accepted"`